### ISSUE TITLE
Assertion error when trying to take jacobian of projected gradient solution using projection_polyhedron

### ISSUE THREAD

**Issue by Basant1861** [google/jax#13089]
### Description

The output of compute_pg is a vector . If I just evaluate f_x below at the desired input it works but the jacobian doesnt seem to work. I am not sure what is happening here.

```
@partial(jit, static_argnums=(0,))
 def proj(self,p,C):
     return projection_polyhedron(p,C,check_feasible = False)

@partial(jit, static_argnums=(0,))
def compute_pg(self,p,cx,cy,x_obs_inp,y_obs_inp,vx_obs_inp, vy_obs_inp):
        
        p = jnp.reshape(p,(jnp.shape(p)[0],1))
        b_obs = jnp.hstack((x_obs_inp+self.p_obs_x_ub,-(x_obs_inp-self.p_obs_x_lb),
                                        y_obs_inp+self.p_obs_y_ub,-(y_obs_inp-self.p_obs_y_lb)))

        b_obs = jnp.reshape(b_obs,(jnp.shape(self.C_obs)[0],1))
        pg = ProjectedGradient(fun=self.compute_obstacle_penalty_temp,projection=self.proj,jit=True)
        pg_sol = pg.run(p,hyperparams_proj=(self.A_obs,self.a_obstacle,self.C_obs,b_obs),c_x=cx,c_y=cy,vx_obs_inp=vx_obs_inp,vy_obs_inp=vy_obs_inp).params
        return pg_sol 

f_x = lambda cx: self.compute_pg(p[0,:],cx,best_cy.T,x_obs_inp[0,:],y_obs_inp[0,:],vx_obs_inp[0,:], vy_obs_inp[0,:])
dp_dcx = jax.jacobian(f_x)(best_cx.T)

```

This is the error I get:

  ```
File "/home/ims/ros2_ws/install/mpc_python/lib/python3.8/site-packages/mpc_python/mpc_expert_bilevel.py", line 592, in compute_bilevel
    dp_dcx = jax.jacobian(f_x)(best_cx.T)
  File "/home/ims/.local/lib/python3.8/site-packages/jaxopt/_src/implicit_diff.py", line 236, in solver_fun_bwd
    vjps = root_vjp(optimality_fun=optimality_fun, sol=sol,
  File "/home/ims/.local/lib/python3.8/site-packages/jaxopt/_src/implicit_diff.py", line 69, in root_vjp
    u = solve(matvec, v)
  File "/home/ims/.local/lib/python3.8/site-packages/jaxopt/_src/linear_solve.py", line 193, in solve_normal_cg
    Ab = rmatvec(b)  # A.T b
  File "/home/ims/.local/lib/python3.8/site-packages/jaxopt/_src/linear_solve.py", line 145, in <lambda>
    return lambda y: transpose(y)[0]
AssertionError
```
What I observed was that if I set implicit_diff=False in the ProjectedGradient then it works but is super slow.Kindly advice. 

### What jax/jaxlib version are you using?

0.3.22

### Which accelerator(s) are you using?

GPU

### Additional system info

3.8.10,Ubuntu 20.04.4 LTS,1th Gen Intel® Core™ i9-11980HK @ 2.60GHz × 16,NVIDIA Corporation / NVIDIA GeForce RTX 3080 Laptop GPU/PCIe/SSE2

### NVIDIA GPU info

-----------------------------------------------------------------------------+
| NVIDIA-SMI 510.85.02    Driver Version: 510.85.02    CUDA Version: 11.6     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ...  Off  | 00000000:01:00.0 Off |                  N/A |
| N/A   56C    P5    21W /  N/A |   1335MiB / 16384MiB |      5%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A      1203      G   /usr/lib/xorg/Xorg                134MiB |
|    0   N/A  N/A      1845      G   /usr/lib/xorg/Xorg                723MiB |
|    0   N/A  N/A      2168      G   /usr/bin/gnome-shell              101MiB |
|    0   N/A  N/A      3875      G   /usr/lib/firefox/firefox          356MiB |
|    0   N/A  N/A    221513      G   gnome-control-center                3MiB |
+-----------------------------------------------------------------------------+

**Comment by jakevdp** [comment]
This looks like a duplicate of #13088; in the future please feel free to just ask once – the issues and discussions are both monitored by the team!
